app/controllers/vendor_api/decisions_controller.rb

```diff
@@ -61,15 +92,25 @@ def confirm_enrolment
   private
 
     def application_choice
-      @application_choice ||= GetApplicationChoicesForProviders.call(providers: current_provider)
-        .find(params[:application_id])
+      @application_choice ||= GetApplicationChoicesForProviders.call(providers: current_provider).find(params[:application_id])
     end
 
     def respond_to_decision(decision)
-      if decision.save
-        render json: { data: SingleApplicationPresenter.new(application_choice).as_json }
-      else
-        render_validation_errors(decision.errors)
+      begin
+        if decision.save
+          render json: { data: SingleApplicationPresenter.new(application_choice).as_json }
+        else
+          render_validation_errors(decision.errors)
+        end
+      rescue ProviderAuthorisation::NotAuthorisedError => e
```

Comment:
As the whole method is rescuable, no need for a `begin`? https://stackoverflow.com/a/1542682